DWB fails when goal is directly opposite of robot along X or Y axis

<!--
For general questions, please ask on ROS answers: https://answers.ros.org, make sure to include at least the `ros2` tag and the rosdistro version you are running, e.g. `ardent`.
For general design discussions, please post on discourse: https://discourse.ros.org/c/ng-ros
Not sure if this is the right repository? Open an issue on https://github.com/ros-planning/navigation2
For Bug report or feature requests, please fill out the relevant category below
-->

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 18.04
- Installation type:
  - source
- Version or commit hash:
  - master
- DDS implementation:
  - default
- Client library (if applicable):
  - rclcpp

#### Steps to reproduce issue
<!-- Detailed instructions on how to reliably reproduce this issue http://sscce.org/
``` code that can be copy-pasted is preferred ``` -->
When robot is located near the center of a pole and the the 2D Nav goal is on the other side of the map near the center of the pole, the local planner fails with the following error msg:
```
[ERROR] [DWBLocalPlanner]: No valid trajectories out of 399! 
[ERROR] [DWBLocalPlanner]: 1.00:   GoalDist/Trajectory Hits Unreachable Area.

```

#### Additional information
![screenshot from 2018-12-07 17-03-16](https://user-images.githubusercontent.com/39755151/49679762-a3576780-fa42-11e8-9a5b-b767b15a4a94.png)

--- a/nav2_dwb_controller/costmap_queue/src/costmap_queue.cpp
+++ b/nav2_dwb_controller/costmap_queue/src/costmap_queue.cpp
@@ -85,6 +85,7 @@ CellData CostmapQueue::getNextCell()
 {
   // get the highest priority cell and pop it off the priority queue
   CellData current_cell = front();
+  pop();
 
   unsigned int index = current_cell.index_;
   unsigned int mx = current_cell.x_;
@@ -107,8 +108,6 @@ CellData CostmapQueue::getNextCell()
     enqueueCell(index + size_x, mx, my + 1, sx, sy);
   }
 
-  // pop once we have our cell info
-  pop();
   return current_cell;
 }
 

